taskfile/ast/prompt.go

```diff
@@ -0,0 +1,21 @@
+package ast
+
+type Prompt []string
+
+func (p *Prompt) UnmarshalYAML(unmarshal func(any) error) error {
+	var m []string
+	err := unmarshal(&m)
+	if err == nil {
+		*p = m
+		return nil
+	}
+
+	var s string
+	err = unmarshal(&s)
+	if err != nil {
+		return err
+	}
+
+	*p = []string{s}
+	return nil
+}
```

Comment:
You should use the same pattern as we have everywhere else (see include.go for example)

```suggestion
func (p *Prompt) UnmarshalYAML(node *yaml.Node) error {
	switch node.Kind {
	case yaml.ScalarNode:
		var str string
		if err := node.Decode(&str); err != nil {
			return errors.NewTaskfileDecodeError(err, node)
		}
		*p = []string{str}
		return nil
	case yaml.SequenceNode:
		var list []string
		if err := node.Decode(&list); err != nil {
			return errors.NewTaskfileDecodeError(err, node)
		}
		*p = list
		return nil
	}
	return errors.NewTaskfileDecodeError(nil, node).WithTypeMessage("prompt")
}

```